### ISSUE TITLE
Download of zip file fails to complete

### ISSUE THREAD

**Issue by tstefanich** [processing/p5.js-web-editor#1137]
#### Nature of issue?

- Found a bug

#### Details about the bug: 

- Web browser and version: Chrome: 76.0.3809.100, Firefox: 67.0 
- Operating System: Mac, Windows
- Steps to reproduce this bug:
Use this sketch as an example, https://editor.p5js.org/FRESHPRNCE/sketches/S1Kqxl-HQ
Go to "File" in the menu bar and click  "Download" and it will stop downloading at about ~676kb, hang there for a few seconds and then fail. Attached is an image of the errors in the console. 

** We used the online editor for a high school program over the summer and 15 out of the 54 student projects were unable to be downloaded. So I have more links I could send but all the errors seem to be the same from sketch to sketch. 

<img width="1679" alt="Screenshot 2019-08-17 17 25 52" src="https://user-images.githubusercontent.com/810727/63218670-4e610a80-c114-11e9-950e-3b3c0efb1e13.png">

**Comment by catarak** [comment]
thanks for reporting this bug! i'm able to reproduce. not sure what's going on here but i can do some digging.

**Comment by tstefanich** [comment]
Thank you for looking into this :)

**Comment by shakti97** [comment]
Can I try this?

**Comment by shakti97** [comment]
I just found the problem for the error, its an action without dispatching something & I don't think it should be the action if its not dispatching something so it should better be a function.
here i am talking about this exportProjectAsZip action in the projectAction. 
So in my mind there are two ways to solve this either by passing type null so that the default reducer will handle it and the other one to remove the exportProjectAsZip from action
or if you can tell me any other way @catarak 

Note:- this will only solve the error shown above in the pic . But not the downloading issue

**Comment by shakti97** [comment]
if you want I can generate a separate issue first describing the action error and then sent a pr by resolving the error @catarak

**Comment by catarak** [comment]
My instinct here is that the bug you found is a red herring—I think maybe the server is returning an error, and the front end is not handling it properly?

**Comment by catarak** [comment]
i was able to get rid of that error, but running the server locally, it seems like this is silently failing, as there are no errors from the server, either. it seems like it's getting stuck somewhere and just hanging!

**Comment by ykabusalah** [comment]
Can I try doing this?

**Comment by everythingability** [comment]
https://www-users.york.ac.uk/~tas509/p5Downloads.png

I get this, failed downloads... My p5js account is full and so I was trying to download a lot of projects. Lots of them fail. Like this one for example

https://editor.p5js.org/tom.smith/sketches/rG16mHfhv

**Comment by catarak** [comment]
Thanks for the additional information!

**Comment by ron-debajyoti** [comment]
Hey @catarak ! As a beginner, I would like to give this a try !

**Comment by ron-debajyoti** [comment]
I tried to replicate the error but it seems to be working for me. I think this has to be the server issue of S3. @catarak Would appreciate your views about this :smile:

**Comment by shubhamkapoor01** [comment]
I would like to give this a try as a beginner!

**Comment by haschdl** [comment]
TL;DR: Could this issue be closed in favour of #2038 

hi @tstefanich, I would like to ask if this issue is still relevant for you and others and if it could closed and superseded by #2038   Downloading a single sketch as a zip file is not working at all (all files end up as zero bytes)! 

@ron-debajyoti  reported that everything is working so it looked like some intermittent problem.

**Comment by tstefanich** [comment]
Hi thanks for checking back in! 

I can't test unfortunately, the old links I was testing (which we had like 20+ links) don't seem to work anymore. 
Here is one of them - http://alpha.editor.p5js.org/full/Bkg6zTkr7

Do you know if these projects are still available/recoverable?

**Comment by ghost** [comment]
This link fails 90%. Well anything over 100mb's has this issue on my end. Using firefox latest ver.
https://github.com/eried/portapack-mayhem/releases/download/v1.5.4/mayhem_v1.5.4_COPY_TO_SDCARD.zip

**Comment by raclim** [comment]
Hi all! I'm going to close this issue for now since it seems like its in a similar vein to https://github.com/processing/p5.js-web-editor/issues/2038, https://github.com/processing/p5.js-web-editor/issues/2057, and https://github.com/processing/p5.js-web-editor/issues/2071. Please feel free to reopen this issue or create a new one if anyone feels that this is distinct from the other issues listed, thanks so much!
